src/date/index.ts

```diff
@@ -27,3 +31,344 @@ export function getMonthAbbrByIndex(monthNumber: number): string {
+ *
+ * @param {number} lastSupportedYear - last year in number from which array of values needs to be returned
+ * @param {number} offsetFromCurrentFinancialYear - offset from current year of the year upto which the values have to be returned
+ *
+ * @example
+ * ```
+ * getReportDateInput(2020, 1);  // Output will be [{startDate: 'Apr 2020', endDate: 'Mar 2021'}]
+ * getReportDateInput(2020);  // Output will be [{startDate: 'Apr 2021', endDate: 'Mar 2022'}, {startDate: 'Apr 2020', endDate: 'Mar 2021'}]
+ * ```
+ */
+export function getReportDateInput(lastSupportedYear: number, offsetFromCurrentFinancialYear: number = 0) {
+  try {
+    const dateInput = [];
+    const currentYear = new Date().getFullYear();
+    const currentMonth = new Date().getMonth();
+    const currentFinancialYear = currentMonth < 4
+      ? currentYear - 1 - offsetFromCurrentFinancialYear
+      : currentYear - offsetFromCurrentFinancialYear;
+
+    for (let financialYear = currentFinancialYear; financialYear >= lastSupportedYear; financialYear--) {
+      const dateEntry = {
+        startDate: `Apr ${financialYear}`,
+        endDate: `Mar ${financialYear + 1}`
+      };
+
+      dateInput.push(dateEntry);
+    }
+
+    return dateInput;
+
+  } catch (error) {
+    console.error('Report-date generation error:', error);
+    return [];
+  }
+}
+
+
+/**
+ * This method inserts "/" while entering dates in input element
+ *
+ * @param {string} inputDate - string entered in input element
+ *
+ * @example
+ * ```
+ * formatDateWithBackSlash('121');  // Output will be '12/1'
+ * formatDateWithBackSlash('12');  // Output will be '12/'
+ * formatDateWithBackSlash('1');  // Output will be '1'
+ * formatDateWithBackSlash('1213');  // Output will be '12/13'
+ * formatDateWithBackSlash('12121212');  // Output will be '12/12/1212'
+ * ```
+ */
+export function formatDateWithBackSlash(inputDate: string) {
+  try {
+    const str = inputDate.replace(/[^0-9]/g, '');
+    const { length } = str;
+    let result = str;
+
+    if (length > 2) {
+      if (length >= 4) {
+        result = str.slice(0, 2) + '/' + str.slice(2, 4) + '/' + str.slice(4, length);
+
+      } else {
+        result = str.slice(0, 2) + '/' + str.slice(2, length);
+      }
+
+    } else if (length === 2) {
+      result = str.slice(0, 2) + '/';
+    }
+
+    return result;
+
+  } catch (error) {
+    console.log('Error in formatDateWithBackSlash: ', error);
+  }
+}
+
+
+/**
+ * This method Checks whether date string is Valid Date of birth or not.
+ * NOTE:- Doesn't check for Minor (Age < 18).
+ *
+ * @param {string} inputDate - string entered in input element (ddmmyyyy or dd/mm/yyyy)
+ *
+ * @example
+ * ```
+ * dobValidationCheck('10/10/2000');  // Output will be true
+ * dobValidationCheck('10/10/1500');  // Output will be false
+ * ```
+ * @return {boolean} result - return true or false
+ * Validations :- isValidDate , Future date, formatted length = 10, age not more than 120.
```

Comment:
what do u mean by isValidDate?

date input is as expected, given date is not of future, .... etc

Reply:
updated with remark